sway-lsp/src/error.rs

```diff
@@ -0,0 +1,25 @@
+use thiserror::Error;
+use tower_lsp::lsp_types::Diagnostic;
+
+#[derive(Debug, Error)]
+pub enum LanguageServerError {
+    #[error(transparent)]
+    DocumentError(#[from] DocumentError),
+
+    #[error("Failed to create build plan. {0}")]
+    BuildPlanFailed(anyhow::Error),
+    #[error("Failed to compile. {0}")]
+    FailedToCompile(anyhow::Error),
+    #[error("Failed to parse document. {:?}", diagnostics)]
```

Comment:
I think just `Failed to parse document` is fine here instead of also printing the diagnostics. The user can already see the warnings and errors reported from the diagnostics in the editor, so printing them to the console as well might make it overly noisy.